Wollah dit is m

Yannickvdv opened the pull request:
Is nog niet compleet natuurlijk. Maar gewoon begin

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/maatwork/bar/8)
<!-- Reviewable:end -->

Commit:
Added Category and Question models/routes. To be reviewed.

Commit:
Begin categories and questions

C-Alexander commented:
|`app.ts:21`|
|---|
Require questions twice?

C-Alexander commented on models/category.ts:
```diff
@@ -0,0 +1,10 @@
+const Logger = require('../models/logger');
+
+
+module.exports.getCategories = function (callback: Function) {
```
Would work but this isnt the correct naming for Postgresql. The correct table name would be "categories". But this works anyhow

C-Alexander commented on routes/categories.ts:
```diff
@@ -0,0 +1,12 @@
+/* GET categories. */
+router.get('/', function(req, res, next) {
+    category.getCategories(function(error, result){
+        res.render('category', {title: 'Category', msg: error, categories: result.rows});
```
You're rendering "category" but there is no actual page called Category

C-Alexander commented on routes/categories.ts:
```diff
@@ -0,0 +1,12 @@
+var category = require('../models/category');
+
+/* GET categories. */
+router.get('/', function(req, res, next) {
```
This is not accessible from an HTML app. 
Needs code such as   <code>res.header("Access-Control-Allow-Origin", "*");</code> (can insert your origin there ...)

C-Alexander commented on routes/questions.ts:
```diff
@@ -0,0 +1,12 @@
+    })
+});
+
+module.exports = router;
```
See comment in categories.js

C-Alexander commented on models/question.ts:
```diff
@@ -0,0 +1,11 @@
+
+
+module.exports.getQuestions = function (questionID, callback) {
+    const sql = 'SELECT * FROM question, answer WHERE question.questionid = answer.questionid AND question.questionID = $1';
```
The naming convention for Postgres is to have the tables as plural. So questions, answers. But fontys taught us differently (as we learned mssql) so this is fine for now.

Yannickvdv replied:
Oh never knew that, at fontys they always told me to use singular.

C-Alexander commented:
In general, why are you sending pages? Don't you just want to send json?

Additionally, we have a folder (/api/) in routes for api functions. Seems more fitting, to me

C-Alexander commented:
*[app.js, line 15 at r1](https://reviewable.io:443/reviews/maatwork/bar/8#-KwbW7qr02sLRb9pQzE0:-KwbW7qr02sLRb9pQzE1:bb7p34n) ([raw file](https://github.com/maatwork/bar/blob/b4f7eb8f8abfc9c1b42560ede949b2d6cf4bdaef/app.js#L15)):*
> ```JavaScript
> var Logger = require('./models/logger');
> var index = require('./routes/index');
> var register = require('./routes/register');
> ```

test

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maatwork/bar/8)*
<!-- Sent from Reviewable.io -->

C-Alexander commented:
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app.js, line 19 at r1](https://reviewable.io:443/reviews/maatwork/bar/8#-KwbVnfaGoBsBnjEO0zu:-KwbVnfaGoBsBnjEO0zv:b-pftd7q) ([raw file](https://github.com/maatwork/bar/blob/b4f7eb8f8abfc9c1b42560ede949b2d6cf4bdaef/app.js#L19)):*
> var questions = require('./routes/questions');
> var categories = require('./routes/questions');

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maatwork/bar/8)*
<!-- Sent from Reviewable.io -->

C-Alexander commented:
pls update

Commit:
Little changes

Commit:
Merge with upstream

C-Alexander reviewed (changes requested):
Please remove app.js

Commit:
Addition of REST POST API call to add category

Commit:
Change from postgressql to sequelize.

C-Alexander commented on routes/api/quizzes.js:
```diff
@@ -1,23 +1,24 @@
 var express = require('express');
 var router = express.Router();
-var category = require('../../models/category').Category;
+var quiz = require('../../models/qui').Quiz;
```
que? Que si, que no, que dingus dingus dingus

C-Alexander commented on routes/api/quizzes.js:
```diff
@@ -1,23 +1,24 @@
 "use strict";
 var express = require('express');
```
delete the JS.

C-Alexander commented on routes/api/quizzes.ts:
```diff
@@ -1,26 +1,27 @@
 var express = require('express');
 var router = express.Router();
-var category = require('../../models/category').Category;
+var quiz = require('../../models/qui').Quiz;
```
Que si, que no

C-Alexander commented on routes/api/quizzes.ts:
```diff
@@ -1,26 +1,27 @@
     res.header("Access-Control-Allow-Origin", "*");
-    category.create({ title: req.body.title ,description: req.body.description});
+    quiz.create({ title: req.body.title ,description: req.body.description})
+        .catch(error => Logger(error))
```
pls specify category

C-Alexander commented on routes/api/quizzes.ts:
```diff
@@ -1,26 +1,27 @@
 var express = require('express');
```
use let's, not var's

C-Alexander reviewed (changes requested):
Please fix the bug, there's some style problems but those can be survived if needed.

Commit:
Merge branch 'master' of https://github.com/Maatwork/bar

# Conflicts:
#	.idea/workspace.xml
#	app.ts
#	db/foreignkeys.ts
 wot

Commit:
Merge branch 'master' of https://github.com/Maatwork/bar

# Conflicts:
#	.idea/workspace.xml
#	app.ts
#	db/foreignkeys.ts
 wot

Commit:
Questions , quiz,

Questions: (Post,Put, Delete) , quiz: (put, expanding of post),
renaming of category to quiz

C-Alexander committed:
Merge branch 'master' into master
